src/password/views.py

```diff
@@ -0,0 +1,214 @@
+#
+# Plantalytics
+#     Copyright (c) 2016 Sapphire Becker, Katy Brimm, Scott Ewing,
+#       Matt Fraser, Kelly Ledford, Michael Limb, Steven Ngo, Eric Turley
+#     This project is licensed under the MIT License.
+#     Please see the file LICENSE in this distribution for license terms.
+# Contact: plantalytics.capstone@gmail.com
+#
+
+import json
+import logging
+import uuid
+import string
+import random
+
+import cassy
+from django.views.decorators.csrf import csrf_exempt
+from common.exceptions import *
+from common.errors import *
+from django.conf import settings
+from django.http import HttpResponse, HttpResponseForbidden, HttpResponseBadRequest, HttpResponseServerError
+from django.core.mail import send_mail
+
+logger = logging.getLogger('plantalytics_backend.login')
+
+
+@csrf_exempt
+def change(request):
+    """
+    Mock auth endpoint to return success with generated token
+    if correct user/pass are passed in.
+    """
+    if request.method != 'POST':
+        return HttpResponseBadRequest()
+
+    data = json.loads(request.body.decode('utf-8'))
+
+    auth_token = data.get('auth_token', '')
+    username = data.get('username', '')
+    new_password = data.get('password', '')
+    old_password = data.get('old', '')
+    reset_token = data.get('token', '')
+
+    logger.info('Request to change password.')
+    try:
+        # User forgot password and reset token is given via email link
+        if reset_token:
+            logger.info('Attempting password reset using reset_token.')
+            reset_name = cassy.verify_auth_token(reset_token)
+            stored_password = cassy.get_user_password(reset_name)
+            if reset_name != username:
+                logger.warn(
+                    "Username associated with reset token (\'{}\') ".format(reset_name)
+                    + "does not match supplied username (\'{}\').".format(username)
+                )
+                raise PlantalyticsLoginException(LOGIN_ERROR)
+            if new_password == stored_password:
+                logger.warn('Invalid new password.')
+                raise PlantalyticsLoginException(LOGIN_ERROR)
```

Comment:
I think here a more descriptive error would be appropriate. Contrast to the previous exception where the username doesn't match (403 -- Forbidden, generic login-error), here they are the appropriate user, just don't have a good new password. (400 -- bad request, bad password choice).